### ISSUE TITLE
pk_cpp is not removed if kernel changes which keeps reporting an error

### ISSUE THREAD

**Issue by gliga** [kokkos/pykokkos#234]
The following example leads to an error:

```
import pykokkos as pk

@pk.workunit
def map_d(i):
    #i += 1                                                                                                                                                                                                                                                                                                                                           
    pass

def main():
    SIZE = 2
    pk.parallel_for(SIZE, map_d)

main()
```

This happens as we do not support `pass` (but that is a side comment only). The key is that this results in an error.

Once I "fix" the kernel and have the resulting (correct) code:
```
import pykokkos as pk

@pk.workunit
def map_d(i):
    i += 1

def main():
    SIZE = 2
    pk.parallel_for(SIZE, map_d)

main()
```

I am still getting the same error as before.  Only after deleting `pk_cpp` in between kernel updates, the error goes away and my example runs properly.

Action: Remove pk_cpp when kernel changes.

Used pykokkos: `2e0b664c2aec1c923c55a7dc5958fc19f0c327cd`
Used command: `./pk pk_example X.py`

**Comment by kennykos** [comment]
Duplicate of #100

**Comment by gliga** [comment]
Great, let's close one of them. And then we should fix the remaining one.

**Comment by IvanGrigorik** [comment]
https://github.com/kokkos/pykokkos/issues/234 closed, this should be addressed

**Comment by kennykos** [comment]
Closed by #415
